Source/SA/RFmx SpecAn/RFmxSpecAn.cs

```diff
@@ -11,65 +12,334 @@ public struct CommonConfiguration
+                specAn.Acp.Configuration.ConfigureOffsetIntegrationBandwidth(offsetString, offsetConfiguration.IntegrationBandwidth_Hz);
+                specAn.Acp.Configuration.ConfigureOffset(offsetString, offsetConfiguration.Frequency_Hz, offsetConfiguration.SideBand, offsetConfiguration.Enabled);
+                specAn.Acp.Configuration.ConfigureOffsetPowerReference(offsetString, offsetConfiguration.PowerReferenceCarrier, offsetConfiguration.PowerReferenceSpecificIndex);
+                specAn.Acp.Configuration.ConfigureOffsetRelativeAttenuation(offsetString, offsetConfiguration.RelativeAttenuation_dB);
+                specAn.Acp.Configuration.ConfigureOffsetRrcFilter(offsetString, offsetConfiguration.RrcFilterEnabled, offsetConfiguration.RrcAlpha);
+            }
+        }
+
+        public static void ConfigureAmpm(RFmxSpecAnMX specAn, AmpmConfiguration ampmConfig, string selectorString = "")
+        {
+            specAn.Ampm.Configuration.SetMeasurementEnabled(selectorString, true);
+            specAn.Ampm.Configuration.SetAllTracesEnabled(selectorString, true);
+            specAn.Ampm.Configuration.ConfigureMeasurementInterval(selectorString, ampmConfig.MeasurementInterval_s);
+            specAn.Ampm.Configuration.ConfigureDutAverageInputPower(selectorString, ampmConfig.DutAverageInputPower_dBm);
+            RFmxSpecAnMXAmpmReferenceWaveformIdleDurationPresent idleDurationPresent = ampmConfig.ReferenceWaveform.IdleDurationPresent ? RFmxSpecAnMXAmpmReferenceWaveformIdleDurationPresent.True : RFmxSpecAnMXAmpmReferenceWaveformIdleDurationPresent.False;
+            specAn.Ampm.Configuration.ConfigureReferenceWaveform(selectorString, ampmConfig.ReferenceWaveform.WaveformData, idleDurationPresent, ampmConfig.SignalType);
+        }
+        #endregion
+
+        #region Measurement Results
+        public static TxpResults FetchTxp(RFmxSpecAnMX specAn, string selectorString = "")
+        {
+            TxpResults txpResults = new TxpResults();
+            specAn.Txp.Results.FetchMeasurement(selectorString, 10.0, out txpResults.AverageMeanPower_dBm,
+                out txpResults.PeakToAverageRatio_dB, out txpResults.MaximumPower_dBm, out txpResults.MinimumPower_dBm);
+            return txpResults;
+        }
+
+        public static AcpResults FetchAcp(RFmxSpecAnMX specAn, string selectorString = "")
+        {
+            double[] lowerRelativePower = null;
+            double[] upperRelativePower = null;
+            double[] lowerAbsolutePower = null;
+            double[] upperAbsolutePower = null;
+            specAn.Acp.Results.FetchOffsetMeasurementArray(selectorString, 10.0,
+                ref lowerRelativePower, ref upperRelativePower, ref lowerAbsolutePower, ref upperAbsolutePower);
+            AcpResults results = new AcpResults()
+            {
+                OffsetResults = new AcpOffsetResults[lowerRelativePower.Length]
+            };
+            for (int i = 0; i < lowerRelativePower.Length; i++)

```

Comment:
we need to iterate over whichever is greater, lowerRelativePower.Length or upperRelativePower.Length

Reply:
They should be equal from my understanding as RFmx fills them with NaN, do you have another experience in this regard?

Reply:
![grafik](https://user-images.githubusercontent.com/53251025/66656577-9137ee00-ec3e-11e9-81f7-a1994c20d504.png)